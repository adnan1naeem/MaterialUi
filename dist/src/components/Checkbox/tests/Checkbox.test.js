"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const react_1 = __importDefault(require("react"));
const enzyme_adapter_react_16_1 = __importDefault(require("enzyme-adapter-react-16"));
const __1 = __importDefault(require("../"));
const enzyme_1 = require("enzyme");
enzyme_1.configure({ adapter: new enzyme_adapter_react_16_1.default() });
test('when title prop is passed', () => {
    const checkbox = enzyme_1.shallow(react_1.default.createElement(__1.default, { label: "hello" }));
    expect(checkbox.props().label).toBeUndefined();
});
test('when value prop is passed false', () => {
    const checkbox = enzyme_1.shallow(react_1.default.createElement(__1.default, { value: false }));
    expect(checkbox.props().value).toBeFalsy();
    expect(checkbox).toMatchSnapshot();
});
test('when indeterminate prop is passed true', () => {
    const checkbox = enzyme_1.shallow(react_1.default.createElement(__1.default, { indeterminate: false }));
    expect(checkbox.props().indeterminate).toBeFalsy();
    expect(checkbox).toMatchSnapshot();
});
// test('when checked prop is passed', () => {
//     const checkbox = shallow(
//         <Checkbox value={false} />
//     );
//     expect(checkbox.props().value).toBeFalsy();
//     expect(checkbox).toMatchSnapshot();
// });
//# sourceMappingURL=Checkbox.test.js.map