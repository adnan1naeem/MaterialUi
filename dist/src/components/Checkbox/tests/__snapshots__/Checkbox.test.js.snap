// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when indeterminate prop is passed true 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BlueRainCheckbox
    indeterminate={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
<<<<<<< HEAD
      "children": <WithStyles(FormControlLabel)
=======
      "children": <WithStyles(WithFormControlContext(FormControlLabel))
>>>>>>> abe0c27cbfbc48ff736e295ad073f2440bae9139
        control={
                <WithStyles(Checkbox)
                        checked={undefined}
                        indeterminate={false}
                        onChange={undefined}
                        value={undefined}
                />
        }
        label={undefined}
/>,
      "row": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "control": <WithStyles(Checkbox)
          checked={undefined}
          indeterminate={false}
          onChange={undefined}
          value={undefined}
/>,
        "label": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
<<<<<<< HEAD
        "children": <WithStyles(FormControlLabel)
=======
        "children": <WithStyles(WithFormControlContext(FormControlLabel))
>>>>>>> abe0c27cbfbc48ff736e295ad073f2440bae9139
          control={
                    <WithStyles(Checkbox)
                              checked={undefined}
                              indeterminate={false}
                              onChange={undefined}
                              value={undefined}
                    />
          }
          label={undefined}
/>,
        "row": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "control": <WithStyles(Checkbox)
            checked={undefined}
            indeterminate={false}
            onChange={undefined}
            value={undefined}
/>,
          "label": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`when value prop is passed false 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BlueRainCheckbox
    value={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
<<<<<<< HEAD
      "children": <WithStyles(FormControlLabel)
=======
      "children": <WithStyles(WithFormControlContext(FormControlLabel))
>>>>>>> abe0c27cbfbc48ff736e295ad073f2440bae9139
        control={
                <WithStyles(Checkbox)
                        checked={false}
                        onChange={undefined}
                        value={undefined}
                />
        }
        label={undefined}
/>,
      "row": true,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "control": <WithStyles(Checkbox)
          checked={false}
          onChange={undefined}
          value={undefined}
/>,
        "label": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
<<<<<<< HEAD
        "children": <WithStyles(FormControlLabel)
=======
        "children": <WithStyles(WithFormControlContext(FormControlLabel))
>>>>>>> abe0c27cbfbc48ff736e295ad073f2440bae9139
          control={
                    <WithStyles(Checkbox)
                              checked={false}
                              onChange={undefined}
                              value={undefined}
                    />
          }
          label={undefined}
/>,
        "row": true,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "control": <WithStyles(Checkbox)
            checked={false}
            onChange={undefined}
            value={undefined}
/>,
          "label": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
